src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -63,6 +63,73 @@ namespace OpenLoco::World::MapGenerator
         }
     }
 
+    static void generateRivers(const S5::Options& options, HeightMap& heightMap)
+    {
+        for (auto i = 0; i < options.numRiverbeds; i++)
+        {
+            const auto riverEastWest = getGameState().rng.randBool();
+            const auto riverWidth = getGameState().rng.randNext(options.minRiverWidth, options.maxRiverWidth);
+            const auto riverbedHeight = options.minLandHeight;
+
+            // We'll be varying the bank width as we meander
+            auto riverbankWidth = options.riverbankWidth;
+            auto totalRiverWidth = riverWidth + 2 * riverbankWidth;
+            auto easternBankOffset = riverWidth + riverbankWidth;
+
+            // Pivot: generate a random X position
+            auto xStartPos = getGameState().rng.randNext(0.15 * kMapColumns, 0.85 * kMapColumns);
+            for (auto yPos = 0; yPos < kMapRows; yPos++)
```

Comment:
Should probably adhere to the heightmap width/height. Hadn't done so because it was 512x512 before, i.e. too large. However, that will no longer be the case after #2599.